qa/SidechainTestFramework/account/smart_contract_resources/contracts/ForgerStakesV2.sol

```diff
@@ -0,0 +1,116 @@
+        bytes1 vrf2;
+        uint256 stakedAmount;
+    }
+
+    //read-write methods
+
+    /*
+      Register a new forger.
+      Vrf key is split in two separate parameters, being longer than 32 bytes.
+      The methods accepts ZEN value: the sent value will be converted to the initial stake assigned to the forger.
+    */
+    function registerForger(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint32 rewardShare, address reward_address) external payable;
+
+     /*
+      Upgrade an existingw forger.
+      Vrf key is split in two separate parameters, being longer than 32 bytes.
+    */
+    function updateForger(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint32 rewardShare, address reward_address, bytes32 signature1, bytes32 signature2) external;
+
+    /*
+      Delegate a stake to a previously registered forger.
+      Vrf key is split in two separate parameters, being longer than 32 bytes.
+    */
+    function delegate(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2) external payable;
+
+    /*
+      Wiithdraw (unstake) a previously assigned stake.
+      Vrf key is split in two separate parameters, being longer than 32 bytes.
+    */
+    function withdraw(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint256 amount) external;
+
+    //read only methods
+
+    /*
+       Returns the total stake amount at the end of one or more consensus epoch assigned to a specific foger.
+       vrf, signKey and delegator are optional: if all are null, the total stake amount will be returned. If only
+         delegator is null, all the stakes assigned to the forger will be summed
+       consensusEpochStart and maxNumOfEpoch are optional: if both null, the data at the  current consensus epoch is returned
+       Return array contains also elements with 0 value. Returned values are ordered by epoch, and the array lenght may
+       be < maxNumOfEpoch if the current consensus epoch is < (consensusEpochStart + maxNumOfEpoch)
+    */
+    function stakeTotal(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, address delegator, uint32 consensusEpochStart, uint32 maxNumOfEpoch) external view returns (uint256[] memory);
+
+    /*
+       Return total sum payd to reward_address  at the end of one or more consensus epochs.
+       Return array contains also elements with 0 value. Returned values are ordered by epoch, and the array lenght may
+       be < maxNumOfEpoch if the current consensus epoch is < (consensusEpochStart + maxNumOfEpoch)
+    */
+    function rewardsReceived(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2, uint32 consensusEpochStart, uint32 maxNumOfEpoch) external view returns (uint256[] memory);
+
+    /*
+    /  Returns the info of a specific registered forger.
+    */
+    function getForger(bytes32 signPubKey, bytes32 vrf1, bytes1 vrf2) external view returns (ForgerInfo memory);
+
+    /*
+    /  Returns the paginated list of all the registered forgers
+    /  Each element of the list is the detail of a specific forger.
```

Comment:
Maybe it can be added that if startIndex == -1, that means there are no other records to be retrieved.
For consistency, or each return param has a name (as `next` below) or just has its type.